### PR TITLE
feat: Bwameth index

### PR DESCRIPTION
**Pull request by tdayris**
### QC

* [x] I confirm that I have followed the [documentation for contributing to `snakemake-wrappers`](https://snakemake-wrappers.readthedocs.io/en/stable/contributing.html).

While the contributions guidelines are more extensive, please particularly ensure that:
* [x] `test.py` was updated to call any added or updated example rules in a `Snakefile`
* [x] `input:` and `output:` file paths in the rules can be chosen arbitrarily
* [x] wherever possible, command line arguments are inferred and set automatically (e.g. based on file extensions in `input:` or `output:`)
* [x] temporary files are either written to a unique hidden folder in the working directory, or (better) stored where the Python function `tempfile.gettempdir()` points to 
* [x] the `meta.yaml` contains a link to the documentation of the respective tool or command under `url:`
* [x] conda environments use a minimal amount of channels and packages, in recommended ordering

## Summary by CodeRabbit

- **New Features**
	- Introduced `environment.linux-64.pin.txt` and `environment.yaml` files for simplified conda environment setup tailored for bioinformatics applications.
	- Added `meta.yaml` file containing essential metadata for the `bwa-meth index` functionality, enhancing user guidance.
	- Implemented a new Snakefile for testing `bwameth` indexing functionality, ensuring reliable performance.
	- Created a new genomic FASTA file for sample data storage and analysis.
	- Developed a wrapper script to automate the BWA-Meth indexing process, improving user experience.

- **Tests**
	- Added a new test function to validate the `bwameth` indexing command, ensuring reliability in processing.

### CODE REVIEW

File: bio/bwameth/index/wrapper.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    subcommand = "index-mem2"
+
+with TemporaryDirectory() as tempdir:
+    # Create symlink to avoid bwa-meth index to be written next to the input reference file
+    ref_basename = os.path.basename(snakemake.input[0])
+    used_reference = os.path.join(tempdir, ref_basename)
+    os.symlink(os.path.abspath(snakemake.input[0]), os.path.join(tempdir, ref_basename))
+
+    # Find user-defined reference directory
+    prefix = os.path.commonprefix([os.path.abspath(f) for f in snakemake.output])
```
Comment (fgvieira) [inline bio/bwameth/index/wrapper.py:30]: I think you can just write:
```suggestion
    prefix = os.path.commonprefix(snakemake.output)
```

File: bio/bwameth/index/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+name: bwa-meth index
+url: https://github.com/brentp/bwa-meth?tab=readme-ov-file#index
```
Comment (fgvieira) [inline bio/bwameth/index/meta.yaml:2]:
```suggestion
url: https://github.com/brentp/bwa-meth
```

File: bio/bwameth/index/wrapper.py (hunk 1 above)
Comment (tdayris) [reply bio/bwameth/index/wrapper.py:30]: It works if and only if the output directory is not `${PWD}`. In that case, the `outdir` variable is equal to nothing (`""`) and the final `mv` fails.

If i change this, then I shall change the `outdir` variable to make it never empty. Should I proceed ?

Comment (fgvieira) [reply bio/bwameth/index/wrapper.py:30]: Then I think it is ok! :smile:

**Review by fgvieira — APPROVED** [review]

**Pull request merged**